### PR TITLE
[DataLayout] Change return type of `getStackAlignment` to `MaybeAlign`

### PR DESCRIPTION
**Pull request by s-barannikov**
Currently, `getStackAlignment` asserts if the stack alignment wasn't specified. This makes it inconvenient to use and complicates testing.

This change also makes `exceedsNaturalStackAlignment` method redundant.

### CODE REVIEW

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/lib/Target/AArch64/AArch64CallingConvention.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ static bool CC_AArch64_Custom_Block(unsigned &ValNo, MVT &ValVT, MVT &LocVT,
   }
 
   const Align StackAlign =
-      State.getMachineFunction().getDataLayout().getStackAlignment();
+      *State.getMachineFunction().getDataLayout().getStackAlignment();
```
Comment (arsenm) [inline llvm/lib/Target/AArch64/AArch64CallingConvention.cpp:213]: This should probably try to handle the missing case?

Comment (s-barannikov) [reply llvm/lib/Target/AArch64/AArch64CallingConvention.cpp:213]: It shouldn't be missing. I'll add an assert (here and in other places).

**Comment by s-barannikov** [comment]
ping?

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**